dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/liberty/CDIProviderService.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.cdi.liberty;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import javax.enterprise.inject.spi.CDIProvider;
+import javax.enterprise.inject.spi.CDI;
+
+import org.osgi.service.component.annotations.Component;
+
+import com.ibm.ejs.util.Util;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.cdi.CDIService;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import io.openliberty.cdi.spi.CDIExtensionMetadata;
+
+/**
+ * This class is to get hold all necessary services.
+ */
+@Component(name = "com.ibm.ws.cdi.liberty.CDIRuntimeImpl", service = {CDIProvider.class}, property = { "service.vendor=IBM" })
+public class CDIProviderService implements CDIProvider {
```

Comment:
This class does not need to be a Component as it's managed by `ServiceLoader` rather than as an OSGi service.

The Javadoc should also be improved to say it's a thin layer which looks up the real `CDIProvider` as an OSGi service.